discord-library/src/main/java/com/blank/humanity/discordbot/commands/AbstractCommand.java

```diff
@@ -157,123 +130,160 @@ public void updateCommandDefinition() {
      * 
      * @param commandData Already initialized CommandData Object. Command name
      *                    and description is already pre-filled.
+     * @param definition  {@link CommandDefinition} for configurable option
+     *                    names and descriptions
      * @return The modified CommandData
      */
-    protected abstract SlashCommandData createCommandData(
-        SlashCommandData commandData);
+    public abstract CommandData createCommandData(SlashCommandData commandData,
+        CommandDefinition definition);
+
+    protected void setUser(BlankUser newUser) {
+        localUser.set(newUser);
+    }
+
+    protected BlankUser getUser() {
+        return localUser.get();
+    }
+
+    protected GenericCommandInteractionEvent getCommandEvent() {
+        return commandEvent.get();
+    }
+
+    protected CommandAutoCompleteInteractionEvent getAutoCompleteEvent() {
+        return autoCompleteEvent.get();
+    }
+
+    protected Member getMember() {
+        return localMember.get();
+    }
+
+    protected void setMember(Member member) {
+        localMember.set(member);
+    }
+
+    protected MessageEmbed[] getUnsentReply() {
+        return localEmbedsToSend.get();
+    }
 
     /**
      * @see AbstractHiddenCommand
      * @return True if command should always be hidden.
      */
-    protected boolean isEphemeral() {
+    public boolean isEphemeral() {
         return false;
```

Comment:
Why is this always false?

Reply:
Sorry, it's late, I realize now it's because it's the abstract class and will be overwritten when needed.